src/App.vue

```diff
@@ -19,32 +19,62 @@ import { ThemeName } from './plugins/vuetify'
 import { useStore } from 'vuex'
 import { useI18n } from 'vue-i18n'
 
+import { fcm } from './firebase'
+import { getToken } from 'firebase/messaging'
+import { VAPID_KEY, notificationType } from '@/lib/constants'
+
 const store = useStore()
 const isSnackbarShowing = computed(() => store.state.snackbar.show)
 
 const showWarningOnAddressesDialog = ref(false)
 
 const notifications = ref<Notifications | null>(null)
+const isNotificationsAllowed = computed(() => {
+  return store.state.options.isAllowNotifications
+})
+const isBackgroundFetchNotification = computed(() => {
+  console.log(
+    '🚀 ~ App.vue:36 ~ isBackgroundFetchNotification ~ isNotificationsAllowed.value:',
+    isNotificationsAllowed.value
+  )
+  return (
+    isNotificationsAllowed.value &&
+    store.state.options.allowNotificationType === notificationType['Background Fetch']
+  )
+})
+const isPushNotification = computed(() => {
+  console.log(
+    '🚀 ~ App.vue:39 ~ isPushNotification ~ isNotificationsAllowed.value :',
+    isNotificationsAllowed.value
+  )
+  return (
+    isNotificationsAllowed.value &&
+    store.state.options.allowNotificationType === notificationType['Push']
+  )
+})
+const myPK = computed(() => {
+  return store.getters.getMyPK
+})
 
 const themeName = computed(() => {
   return store.state.options.darkTheme ? ThemeName.Dark : ThemeName.Light
 })
 
-const { locale } = useI18n()
-
-onMounted(() => {
-  const instance = getCurrentInstance()
-
-  if (instance) {
-    const notifications = new Notifications(instance.proxy)
-    notifications.start()
+watch(myPK, (newVal, oldVal) => {
+  if (newVal && !oldVal) {
+    const channel = new BroadcastChannel('adm_notifications')
+    channel.postMessage({ isCheckPK: true })
+    channel.onmessage = async (event) => {
+      const data = event.data
+      if (data && data.isNoPrivateKey) {
+        const privateKey = newVal
+        if (privateKey) channel.postMessage({ privateKey })
+      }
+    }
```

Comment:
Seems a little over engineering. Did you try just watching the value of `store.state.passphrase` from Vuex?

```js
watch(() => store.state.passphrase, () => {
  channel.postMessage(passphrase)
})
```

Note: You can retrieve the privateKey and publicKey from passphrase.